includes/SanctionsHooks.php

```diff
@@ -236,24 +236,32 @@ public static function onHistoryRevisionTools( $rev, &$links ) {
 	}
 
 	/**
-	 * @param BaseTemplate $baseTemplate The BaseTemplate base skin template.
-	 * @param array &$toolbox An array of toolbox items.
+	 * @param Skin $skin Skin object
+	 * @param array[] &$sidebar An array of arrays of sidebar items.
 	 */
-	public static function onBaseTemplateToolbox( BaseTemplate $baseTemplate, array &$toolbox ) {
-		$user = $baseTemplate->getSkin()->getRelevantUser();
-		if ( $user ) {
-			$rootUser = $user->getName();
-
-			$toolbox = wfArrayInsertAfter(
-				$toolbox,
-				[ 'sanctions' => [
-					'text' => wfMessage( 'sanctions-link-on-user-page' )->text(),
-					'href' => Skin::makeSpecialUrlSubpage( 'Sanctions', $rootUser ),
-					'id' => 't-sanctions'
-				] ],
-				isset( $toolbox['blockip'] ) ? 'blockip' : 'log'
-			);
+	public static function onSidebarBeforeOutput( Skin $skin, array &$sidebar ) {
+		$user = $skin->getRelevantUser();
+
+		if ( !$user ) {
+		   return;
 		}
+
+		$rootUser = $user->getName();
+
+		$sanctionsLink = [
+			'sanctions' => [
+				'text' => $skin->msg( 'sanctions-link-on-user-page' )->text(),
+				'href' => $skin::makeSpecialUrlSubpage( 'Sanctions', $rootUser ),
+				'id' => 't-sanctions'
+			]
+		];
+
+		$toolbox = $sidebar['TOOLBOX'];
```

Comment:
```suggestion
		if ( !isset( $sidebar['TOOLBOX'] ) ) {
			$sidebar['TOOLBOX'] = [];
		}
		$toolbox = $sidebar['TOOLBOX'];
```
Just for compatibility; I haven't tested MW 1.35, but this line throws below on 1.34:

> <b>Fatal error</b>:  Uncaught TypeError: Argument 1 passed to wfArrayInsertAfter() must be of the type array, null given, called in /srv/femiwiki.com/extensions/Sanctions/includes/SanctionsHooks.php on line 266 and defined in /srv/femiwiki.com/includes/GlobalFunctions.php:207
--
  | Stack trace:
  | # 0 /srv/femiwiki.com/extensions/Sanctions/includes/SanctionsHooks.php(266): wfArrayInsertAfter(NULL, Array, 'log')
  | # 1 /srv/femiwiki.com/includes/Hooks.php(174): SanctionsHooks::onSidebarBeforeOutput(Object(SkinFemiwiki), Array)
  | # 2 /srv/femiwiki.com/includes/Hooks.php(202): Hooks::callHook('SidebarBeforeOu...', Array, Array, NULL)
  | # 3 /srv/femiwiki.com/includes/skins/Skin.php(1355): Hooks::run('SidebarBeforeOu...', Array)
  | # 4 /srv/femiwiki.com/includes/skins/SkinTemplate.php(452): Skin-&gt;buildSidebar()
  | # 5 /srv/femiwiki.com/includes/skins/SkinTemplate.php(215): SkinTemplate-&gt;prepareQuickTemplate()
  | # 6 /srv/femiwiki.com/includes/OutputPage.php(2574): SkinTemplate-&gt;outputPage()
  | # 7 /srv/femiwiki.com/includes/exception/MWExceptionRenderer.php(134): Output in <b>/srv/femiwiki.com/includes/GlobalFunctions.php</b> on line <b>207</b><br />

Reply:
Ohh, yes. I forgot that you need to support < 1.35 MW versions. The hook is not new but it was  modified in 1.35, so previously that index was not set, that's not the only chnage to it though. I am not sure that what you suggest would work, (I know it would in >= 1.35) but I can't get it tested on 1.34 now. 

Can you confirm it works for you in 1.34? If yes, I will amend the patch, otherwise to retain support for both versions you need to use both hooks conditionally.  (That's what has been done in some extensions)

Reply:
Of course, I can. Actually, I already did during the review. So you don't have to worry :smile:

Reply:
OK, done